core/jobs/src/main/java/com/cognifide/aet/job/common/utils/ParamsHelper.java

```diff
@@ -0,0 +1,122 @@
+/**
+ * Automated Exploratory Tests
+ *
+ * Copyright (C) 2013 Cognifide Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.cognifide.aet.job.common.utils;
+
+
+import com.cognifide.aet.job.api.exceptions.ParametersException;
+import java.util.Map;
+import java.util.regex.Pattern;
+import java.util.regex.PatternSyntaxException;
+import javax.xml.xpath.XPath;
+import javax.xml.xpath.XPathExpression;
+import javax.xml.xpath.XPathExpressionException;
+import javax.xml.xpath.XPathFactory;
+import org.apache.commons.lang3.StringUtils;
+
+public final class ParamsHelper {
+
+  private ParamsHelper() {
+    // private constructor for utility class
+  }
+
+  /***
+   * @param key property name that should sore textual representation of Integer
+   * @param params map of parameters
+   * @return integer or null if there is no key in params map
+   * @throws ParametersException if value under key is provided but can not be parsed into Integer
+   */
+  public static Integer getParamAsInteger(String key, Map<String, String> params) throws ParametersException {
+    Integer result = null;
+    if (params.containsKey(key)) {
+      try {
+        result = Integer.parseInt(params.get(key));
+      } catch (NumberFormatException e) {
+        throw new ParametersException(
+            "Provided line: " + params.get(result) + " is not a numeric value.", e);
+      }
+    }
+    return result;
+  }
+
+
+  public static String getParamAsString(String key, Map<String, String> params) {
+    return params.containsKey(key) ? params.get(key) : null;
+  }
+
+  /***
+   * @param values array of objects
+   * @throws ParametersException if all passed objects are null
+   */
+  public static void atLeastOneIsProvided(Object... values)
```

Comment:
How about: [`ObjectUils.firstNotNull`](http://commons.apache.org/proper/commons-lang/javadocs/api-3.1/org/apache/commons/lang3/ObjectUtils.html#firstNonNull%28T...%29) `!= null`?

If you don't like it, please just least `return` instead line `70` and throw the exception without checking `allNull`.